perceval/backends/core/github.py

```diff
@@ -120,83 +148,133 @@ def fetch(self, from_date=DEFAULT_DATETIME):
+        raw_org = next(cache_items)
+        return self.__get_user_and_organization(user_login, raw_user, raw_org)
+
+    def __fetch_assignee_from_cache(self, cache_items):
+        """Fetch issue assignee from cache"""
+
+        raw_assignee = next(cache_items)
+        raw_user = next(cache_items)
+        raw_org = next(cache_items)
+        assignee = self.__get_user_and_organization(raw_assignee['login'], raw_user, raw_org)
+
+        return assignee
+
+    def __fetch_comments_from_cache(self, cache_items):
+        """Fetch issue comments from cache"""
+
+        raw_content = next(cache_items)
+        comments = json.loads(raw_content)
+        for c in comments:
+            tag = next(cache_items)
+            raw_user = next(cache_items)
+            raw_org = next(cache_items)
+            c["user_data"] = self.__get_user_and_organization(c['user']['login'], raw_user, raw_org)
+
+        return comments
+
     @metadata
     def fetch_from_cache(self):
         """Fetch the issues from the cache.
-
         It returns the issues stored in the cache object provided during
         the initialization of the object. If this method is called but
         no cache object was provided, the method will raise a `CacheError`
         exception.
-
         :returns: a generator of items
-
         :raises CacheError: raised when an error occurs accessing the
             cache
         """
         if not self.cache:
             raise CacheError(cause="cache instance was not provided")
 
         cache_items = self.cache.retrieve()
-
+        current_issue = None
         issues = None
 
         while True:
             try:
                 raw_item = next(cache_items)
+
+                if raw_item == '{ISSUES}':
+                    issues = self.__fetch_issues_from_cache(cache_items)
+
+                    if issues:
+                        current_issue = next(issues)
+                        self.__init_extra_issue_fields(current_issue)
+
+                elif raw_item == '{USER}':
+                    current_issue["user_data"] = \
+                        self.__fetch_user_and_organization_from_cache(current_issue['user']['login'], cache_items)
+
+                elif raw_item == '{ASSIGNEE}':
+                    assignee = self.__fetch_assignee_from_cache(cache_items)
+                    current_issue["assignee_data"] = assignee
+
+                elif raw_item == '{COMMENTS}':
+                    comments = self.__fetch_comments_from_cache(cache_items)
+                    current_issue["comments_data"] = comments
+
+                elif raw_item == '{}':
+                    yield current_issue
+                    current_issue = next(issues)
+                    self.__init_extra_issue_fields(current_issue)
+
             except StopIteration:
                 if issues:
-                    for issue in self.__build_issues(issues):
-                        yield issue
+                    (i for i in issues)
```

Comment:
What is the rationale behind this?

Reply:
My mistake, this part can be replaced by 
````python
except StopIteration:
    break
````